samples/charts/data-chart/custom-editing-data/src/index.ts

```diff
@@ -0,0 +1,230 @@
+
+        this.lineXAxis = document.getElementById('LineXAxis') as IgcCategoryXAxisComponent;
+        this.lineYAxis = document.getElementById('LineYAxis') as IgcNumericYAxisComponent;
+
+        this.scatterXAxis = document.getElementById('ScatterXAxis') as IgcNumericXAxisComponent;
+        this.scatterYAxis = document.getElementById('ScatterYAxis') as IgcNumericYAxisComponent;
+
+        var lineSeries = document.getElementById("lineSeries") as IgcLineSeriesComponent;    
+        var pointSeries = document.getElementById("pointSeries") as IgcPointSeriesComponent;
+        var scatterLineSeries = document.getElementById("scatterLineSeries") as IgcScatterLineSeriesComponent;
+        var scatterSeries = document.getElementById("scatterSeries") as IgcScatterSeriesComponent;
+
+        this.lineXAxis.dataSource = this.lineData;
+        lineSeries.dataSource = this.lineData;
+        pointSeries.dataSource = this.lineData;
+        
+        scatterLineSeries.dataSource = this.scatterData;
+        scatterSeries.dataSource = this.scatterData;
+
+        this.lineChart.seriesMouseLeftButtonDown = this.onLineSeriesMouseLeftButtonDown.bind(this);
+        this.lineChart.plotAreaMouseOver = this.onLineChartPlotMouseMove.bind(this);        
+        this.lineChart.plotAreaMouseLeftButtonUp = this.onLineChartPlotMouseLeftButtonUp.bind(this);
+
+        this.scatterChart.seriesMouseLeftButtonDown = this.onScatterLineSeriesMouseLeftButtonDown.bind(this);
+        this.scatterChart.plotAreaMouseOver = this.onScatterChartPlotAreaMouseMove.bind(this);
+        this.scatterChart.plotAreaMouseLeftButtonUp = this.onScatterChartPlotAreaMouseLeftButtonUp.bind(this);
+    }
+
+    public getLineData(): any[]{
+        var lineData: any[] = [
+            { Category: "2010", DataValue: 20, EditingValue: null },
+            { Category: "2011", DataValue: 40, EditingValue: null },
+            { Category: "2012", DataValue: 30, EditingValue: null },
+            { Category: "2013", DataValue: 50, EditingValue: null },
+            { Category: "2014", DataValue: 40, EditingValue: null },
+            { Category: "2015", DataValue: 60, EditingValue: null },
+            { Category: "2016", DataValue: 30, EditingValue: null },
+            { Category: "2017", DataValue: 50, EditingValue: null },
+            { Category: "2018", DataValue: 40, EditingValue: null },
+            { Category: "2019", DataValue: 70, EditingValue: null },
+            { Category: "2020", DataValue: 40, EditingValue: null },
+            { Category: "2021", DataValue: 60, EditingValue: null },
+            { Category: "2022", DataValue: 50, EditingValue: null },
+            { Category: "2023", DataValue: 70, EditingValue: null },
+            { Category: "2024", DataValue: 60, EditingValue: null },
+            { Category: "2025", DataValue: 80, EditingValue: null },
+            { Category: "2026", DataValue: 70, EditingValue: null }
+        ];
+
+        return lineData;
+    }
+
```

Comment:
move getLineData() and getScatterData() to a new file called EditableDataSource.ts with a new class: ```export class EditableDataSource``` and then make those methods static
```
public static getLineData(): any[]
public static getScatterData(): any[]
```
so you can import the EditableDataSource class and use them in this sample:
```
this.lineData = EditableDataSource.getLineData();
this.scatterData = EditableDataSource.getScatterData();
```
this way data will be separated from logic of the sample while displayed in Code Viewer